### PR TITLE
ONNX support

### PR DESCRIPTION
**Pull request by oyamay**
These commits add `lbann.onnx` sub-package to the Python frontend.
This package enables users to convert LBANN models to ONNX models, and vice versa.

* Some test suites are available to check the correctness of the conversions [`scripts/proto/lbann/onnx/tests`](https://github.com/oyamay/lbann/blob/onnx/scripts/proto/lbann/onnx/tests)
* Executable Python scripts are available to convert models on command-line: 
   * LBANN->ONNX: [`scripts/proto/examples/onnx/lbann2onnx.py`](https://github.com/oyamay/lbann/blob/onnx/scripts/proto/examples/onnx/lbann2onnx.py)
   * ONNX->LBANN: [`scripts/proto/examples/onnx/onnx2lbann.py`](https://github.com/oyamay/lbann/blob/onnx/scripts/proto/examples/onnx/onnx2lbann.py)
* The current support status is listed in [`scripts/proto/docs/onnx/support_status.md`](https://github.com/oyamay/lbann/blob/onnx/scripts/proto/docs/onnx/support_status.md).

### CODE REVIEW

File: scripts/proto/lbann/proto.py [hunk 1]
```diff
@@ -170,9 +170,14 @@ def export_proto(self):
...
         return proto
 
+    # Method for returning the names of all fields.
+    def get_field_names(self):
+        return field_names
+
     # Create sub-class.
     return type(type_name, (Layer,),
-                {'__init__': __init__, 'export_proto': export_proto})
+                {'__init__': __init__, 'export_proto': export_proto,
```
Comment (ndryden) [inline scripts/proto/lbann/proto.py]: I reorganized `proto.py` recently, so I think there will be some minor merge conflicts in this file. It should be a straightforward fix, but let me know if you need help.

File: scripts/proto/setup.py [hunk 2]
```diff
@@ -22,5 +22,11 @@ def getLBANNVersion():
     author="Lawrence Livermore National Security, LLC.",
     license="Apache 2.0",
     packages=["lbann"],
-    install_requires=["protobuf>=3.0.0"]
+    install_requires=["protobuf>=3.0.0",
```
Comment (ndryden) [inline scripts/proto/setup.py]: Do we need both protobuf version specs?

Comment (oyamay) [reply scripts/proto/setup.py]: Fixed in oyamay/lbann@1dbbf40.

Comment (oyamay) [reply scripts/proto/setup.py]: Fixed in oyamay/lbann@5ce57a5.

File: scripts/proto/lbann/proto.py (hunk 1 above)
Comment (oyamay) [reply scripts/proto/lbann/proto.py]: Fixed in oyamay/lbann@5ce57a5.

**Pull request merged**